src/node.js

```diff
@@ -388,17 +389,17 @@
       tickDone();
     }
 
+    function PObj(c) {
+      this.callback = c;
+      this.domain = process.domain || null;
+    }
+
     function nextTick(callback) {
       // on the way out, don't bother. it won't get fired anyway.
       if (process._exiting)
         return;
 
-      var obj = {
-        callback: callback,
-        domain: process.domain || null
-      };
```

Comment:
@bnoordhuis I think it has to do with accessing the global `process` object. Though I'm not sure why it would have affect with only the object literal implementation.